manifests/capsule.pp

```diff
@@ -10,31 +10,67 @@
 # $certs_tar::               path to tar file with certs to generate. REQUIRED
 #
 class certs::capsule (
-  $parent_fqdn                  = $fqdn,
-  $capsule_fqdn                 = $certs::node_fqdn,
-  $certs_tar                    = $certs::params::certs_tar
+
+  $parent_fqdn  = $fqdn,
+  $capsule_fqdn = $certs::node_fqdn,
+  $certs_tar    = $certs::params::certs_tar,
+  $generate     = $certs::generate,
+
   ) inherits certs::params {
+  
+  $capsule_client_cert_name = "${capsule_fqdn}-capsule-client"
+  $capsule_client_cert = "${certs::pki_dir}/certs/${capsule_client_cert_name}.crt"
+  $capsule_client_key = "${certs::pki_dir}/private/${capsule_client_cert_name}.key"
+  $capsule_client_bundle = "${certs::pki_dir}/private/${capsule_client_cert_name}.pem"
+
+  cert { $capsule_client_cert_name:
+    hostname      => $capsule_fqdn,
+    purpose       => client,
+    country       => $::certs::country,
+    state         => $::certs::state,
+    city          => $::certs::sity,
+    org           => 'FOREMAN',
+    org_unit      => 'Capsule',
+    expiration    => $::certs::expiration,
+    generate      => $generate,
+    regenerate    => $regenerate,
+    deploy        => $deploy,
+    ca            => $::certs::server_ca,
+    password_file => $certs::ca_key_password_file,
+  }
+
+  Cert[$capsule_client_cert_name] ~>
+  key_bundle { $capsule_client_bundle:
+    key_pair => Cert[$capsule_client_cert_name],
+  } ~>
+  file { $capsule_client_bundle:
+    ensure => file,
+    owner  => $::certs::group,
+    mode   => '0644',
+  }
 
-  # until we support again pushing the cert rpms to the Katello,
-  # make sure the certs_tar path is present
-  validate_present($certs_tar)
-  validate_present($capsule_fqdn)
+  if $generate {
+    # until we support again pushing the cert rpms to the Katello,
+    # make sure the certs_tar path is present
+    validate_present($certs_tar)
```

Comment:
when installing katello bin/katello-installer exec in katello installer, validation fails.